src/snap.c

```diff
@@ -23,19 +23,22 @@
 
 #include "utils.h"
 
-bool verify_appname(const char *appname)
+bool verify_executable_name(const char *executable_name)
 {
-	// snappy appname is of form:
-	// snap.<name>.<app>
+	// The executable name is of form:
+	// snap.<name>.(<appname>|hook.<hookname>)
 	// - <name> must start with lowercase letter, then may contain
 	//   lowercase alphanumerics and '-'
-	// - <app> may contain alphanumerics and '-'
-	const char *whitelist_re = "^snap\\.[a-z][a-z0-9-]*\\.[a-zA-Z0-9-]+$";
+	// - <appname> may contain alphanumerics and '-'
+	// - <hookname must start with a lowercase letter, then may
+	//   contain lowercase letters and '-'
+	const char *whitelist_re =
+	    "^snap\\.[a-z](-?[a-z0-9])*\\.([a-zA-Z0-9](-?[a-zA-Z0-9])*|hook\\.[a-z](-?[a-z])*)$";
 	regex_t re;
 	if (regcomp(&re, whitelist_re, REG_EXTENDED | REG_NOSUB) != 0)
 		die("can not compile regex %s", whitelist_re);
 
-	int status = regexec(&re, appname, 0, NULL, 0);
```

Comment:
Just thinking about something nasty being done. Can we require that total length of `security_tag` be smaller than some sane default. I don't want to let people explode the system by installing a snap with a long app name that is expensive to check.

If there is something in snapd, let's use that limit, if not then let's make up one now.

@jdstrand ^^ FYI